src/arkade-lightning.ts

```diff
@@ -466,37 +468,35 @@ export class ArkadeLightning {
         const address = await this.wallet.getAddress();
         if (!address) throw new Error("Failed to get ark address from wallet");
 
-        // validate we are using a x-only receiver public key
-        let receiverXOnlyPublicKey =
-            await this.wallet.identity.xOnlyPublicKey();
-        if (receiverXOnlyPublicKey.length == 33) {
-            receiverXOnlyPublicKey = receiverXOnlyPublicKey.slice(1);
-        } else if (receiverXOnlyPublicKey.length !== 32) {
-            throw new Error(
-                `Invalid receiver public key length: ${receiverXOnlyPublicKey.length}`
-            );
-        }
+        // validate we are using a x-only public key
+        const ourXOnlyPublicKey = this.normalizeToXOnlyPublicKey(
```

Comment:
normalize func is maybe overkill here, we can admit identity returns a 32 bytes xOnlyKey safely IMO